orc/src/main/java/org/apache/iceberg/orc/IcebergColumn.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.orc;
+
+import java.util.Objects;
+
+/**
+ * Representation of an Iceberg column in ORC metadata mapping.
+ */
+final class IcebergColumn {
+
+  static final String SEPARATOR = ":";
+
+  private final int id;
+  private final boolean required;
+
+  private IcebergColumn(int id, boolean isRequired) {
+    this.id = id;
+    this.required = isRequired;
+  }
+
+  public boolean isRequired() {
+    return required;
+  }
+
+  public int id() {
+    return id;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(this.id);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (obj instanceof org.apache.iceberg.orc.IcebergColumn) {
```

Comment:
Nit: please add empty lines after if and for statements.

Reply:
Done.